src/redux/books/booksSlice.js

```diff
@@ -1,40 +1,51 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { getBooks } from '../../api';
 
-const initialState = [
-  {
-    item_id: 1,
-    title: 'The Great Gatsby',
-    author: 'John Smith',
-    category: 'Fiction',
+const fetchBooks = createAsyncThunk(
+  'books/fetch',
+  async () => {
+    const data = await getBooks();
+    const books = data;
+    return books;
   },
-  {
-    item_id: 2,
-    title: 'Anna Karenina',
-    author: 'Leo Tolstoy',
-    category: 'Fiction',
-  },
-  {
-    item_id: 3,
-    title: 'The Selfish Gene',
-    author: 'Richard Dawkins',
-    category: 'Nonfiction',
-  },
-];
+);
+
+const initialState = {
+  books: [],
+  loading: true,
+  error: false,
+};
 
 const booksSlice = createSlice({
   name: 'books',
   initialState,
   reducers: {
-    add(state, action) {
-      state.push(action.payload);
+    setError(state, action) {
+      // eslint-disable-next-line no-param-reassign
+      state.error = action.payload;
     },
-    remove(state, action) {
-      const index = state.findIndex((state) => state.item_id === action.payload);
-      state.splice(index, 1);
+    setLoading(state, action) {
+      // eslint-disable-next-line no-param-reassign
+      state.loading = action.payload;
     },
   },
+  extraReducers(builder) {
+    builder
+      .addCase(fetchBooks.fulfilled, (state, action) => {
+        // eslint-disable-next-line no-param-reassign
+        state.loading = false;
+        // eslint-disable-next-line no-param-reassign
+        state.books = action.payload;
+      })
+      .addCase(fetchBooks.rejected, (state) => {
+        // eslint-disable-next-line no-param-reassign
+        state.loading = false;
+        // eslint-disable-next-line no-param-reassign
+        state.error = true;
+      });
+  },
 });
```

Comment:
@HAMMAS-SALEEM Thanks for your comments, but redux createSlice uses immerJS  on the back, so even if it looks like mutable code immutable logic is happening thanks to redux and immerJS. They ask you explicitly to do it that way on the official documentation [link to redux async documentation](https://redux.js.org/tutorials/essentials/part-5-async-logic)

Reply:
Hey
I am getting what you are saying but the Linters are supposed to be followed for now to make the Linter test pass you can follow the same approach which I have suggested or you can use your own because Linters issue is the most crucial thing.

Reply:
@HAMMAS-SALEEM Thanks a lot for the feedback, I've achieved to change all the mutable logic into immutable